datumaro/plugins/splitter.py

```diff
@@ -0,0 +1,480 @@
+# Copyright (C) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import logging as log
+import numpy as np
+
+from datumaro.components.dataset import Dataset
+from datumaro.components.extractor import (Transform, AnnotationType, \
+    DEFAULT_SUBSET_NAME)
+
+class TaskSpecificSplitter(Transform):
+    def __init__(self, dataset, seed):
+        super().__init__(dataset)
+
+        np.random.seed(seed)
```

Comment:
I think, this should be done closer to the actual use, because a lot can happen after constructor is called.

Reply:
I'll change this too.

Reply:
Random shuffling occurs several times. 
Do you think I really need to set seed whenever I use the shuffling function?

Reply:
Just make sure the results are reproducible. I suggest moving seed setting as much close to the point of use, as possible.

Reply:
Then, I'll add test codes to check the results are reproducible.